test/Microsoft.ML.TestFrameworkCommon/Utility/Kernel32.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Taken from https://github.com/mellinoe/nativelibraryloader/blob/586f9738ff12688df8f0662027da8c319aee3841/NativeLibraryLoader/Kernel32.cs
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.TestFrameworkCommon.Utility
+{
+    internal static class Kernel32
+    {
+        [DllImport("kernel32")]
```

Comment:
>DllImport [](http://example.com/codeflow?start=9&length=9)

should we specify CharSet = CharSet.Unicode attribute paramter?

Reply:
You would know better than me so you tell me :) I took the code from the gitub link mentioned on top of the file and left it as is, so I am not sure if it should be there or not honestly.

Reply:
I would prefer to use it as the one defined here https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.LoadLibrary.cs. This will avoid converting the input string to ascii and will ensure calling the `W` version of the API.

Reply:
If I remember correctly Eric St. John and I looked at that, and its not included in net core 2.1, which is why we went with this route. Or it may have been a different way to check if a dll is present or not. I can't remember if it was this specific one or not. Are you just suggesting to replace the `DllImport` we have here with how you have it defined in the link you shared?

This is _only_ used during tests to see if a native dll exists and not used in any of the actual production code. If the native dll exists we run the tests, if it doesn't exist then we skip the tests.

Reply:
> If I remember correctly Eric St. John and I looked at that, and its not included in net core 2.1, which is why we went with this route. 

what you mean by "its not included in net core 2.1"?

> This is only used during tests to see if a native dll exists and not used in any of the actual production code. If the native dll exists we run the tests, if it doesn't exist then we skip the tests.

That is fine to keep what you have if this is a test code only. just FYI, we have [NativeLibrary](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativelibrary?view=net-5.0) if you want to do it in more reliable way :-)

Reply:
Ah, _thats_ what Eric St. John and I looked at first. It only supports from .net core 3 and new, and we need to support .net core 2 as well as .net framework, so we couldn't use that and so we went with this other approach.